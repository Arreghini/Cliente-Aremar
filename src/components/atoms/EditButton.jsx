// src/components/atoms/EditButton.jsx
import React from 'react';
import { Pencil } from 'lucide-react';
import PropTypes from 'prop-types';

const EditButton = ({ onEdit, reservationId }) => {
  return (
    <button
      onClick={() => onEdit(reservationId)}
      className="p-2 rounded-full hover:bg-blue-100 text-blue-600 transition-colors"
      title="Editar reserva"
    >
      <Pencil className="w-5 h-5" />
    </button>
  );
};

EditButton.propTypes = {
  onEdit: PropTypes.func.isRequired,
  reservationId: PropTypes.number.isRequired,
};

export default EditButton;
