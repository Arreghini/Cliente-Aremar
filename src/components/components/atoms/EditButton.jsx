import React from 'react';
import reservationService from '../../services/ReservationService';
import { useAuth0 } from "@auth0/auth0-react";
import { FaEdit } from "react-icons/fa";

const EditButton = ({ reservationId, onEdit }) => {
  const { getAccessTokenSilently } = useAuth0();
 
  const handleEdit = async () => {
    try {
      const token = await getAccessTokenSilently();
      console.log('ID de reserva a editar:', reservationId);
      const reservationData = await reservationService.getReservation(token, parseInt(reservationId));
      onEdit(reservationData);
    } catch (error) {
      console.error('Error al obtener la reserva:', error);
    }
  };
  
  return (
    <button 
      onClick={handleEdit} 
      className="text-blue-500 hover:text-blue-700"
    >
      <FaEdit />
    </button>
  );
};

export default EditButton;
