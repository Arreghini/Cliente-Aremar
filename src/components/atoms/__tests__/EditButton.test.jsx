// src/components/atoms/__tests__/EditButton.test.jsx
import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import EditButton from '../EditButton'

// 游릱 Grupo de tests
describe('EditButton', () => {
  it('renderiza el bot칩n con el icono y atributos correctos', () => {
    const mockFn = vi.fn()
    render(<EditButton onEdit={mockFn} reservationId="P3D2" />)

    const button = screen.getByRole('button', { name: /editar reserva/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveAttribute('type', 'button')
    expect(button).toHaveAttribute('title', 'Editar reserva')
    expect(button).toHaveClass('p-2 rounded-full hover:bg-blue-100 text-blue-600 transition-colors')

    // El icono Pencil deber칤a estar en el DOM
    expect(button.querySelector('svg')).toBeInTheDocument()
  })

  it('ejecuta la funci칩n onEdit con el reservationId (string) al hacer clic', () => {
    const mockFn = vi.fn()
    render(<EditButton onEdit={mockFn} reservationId="P3D2" />)

    const button = screen.getByRole('button', { name: /editar reserva/i })
    fireEvent.click(button)

    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenCalledWith('P3D2')
  })

  it('ejecuta la funci칩n onEdit con el reservationId (number) al hacer clic', () => {
    const mockFn = vi.fn()
    render(<EditButton onEdit={mockFn} reservationId={123} />)

    const button = screen.getByRole('button', { name: /editar reserva/i })
    fireEvent.click(button)

    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenCalledWith(123)
  })

  it('no dispara un submit si est치 dentro de un formulario', () => {
    const mockFn = vi.fn()
    const handleSubmit = vi.fn((e) => e.preventDefault())

    render(
      <form onSubmit={handleSubmit}>
        <EditButton onEdit={mockFn} reservationId="TEST123" />
      </form>
    )

    const button = screen.getByRole('button', { name: /editar reserva/i })
    fireEvent.click(button)

    expect(mockFn).toHaveBeenCalledWith('TEST123')
    expect(handleSubmit).not.toHaveBeenCalled()
  })

  it('coincide con el snapshot', () => {
    const { container } = render(<EditButton onEdit={() => {}} reservationId="SNAPSHOT" />)
    expect(container).toMatchSnapshot()
  })
})
