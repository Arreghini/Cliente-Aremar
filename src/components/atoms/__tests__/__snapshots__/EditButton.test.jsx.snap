// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditButton > coincide con el snapshot 1`] = `
<div>
  <button
    class="p-2 rounded-full hover:bg-blue-100 text-blue-600 transition-colors"
    title="Editar reserva"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-pencil w-5 h-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.174 6.812a1 1 0 0 0-3.986-3.987L3.842 16.174a2 2 0 0 0-.5.83l-1.321 4.352a.5.5 0 0 0 .623.622l4.353-1.32a2 2 0 0 0 .83-.497z"
      />
      <path
        d="m15 5 4 4"
      />
    </svg>
  </button>
</div>
`;
